File: apps/api/src/schemas/watch.py
from pydantic import BaseModel
from typing import Optional


class WatchCreateSchema(BaseModel):
    name: str
    question: str
    aoi: dict
    sensor_preference: str = "auto"
    frequency: str = "once"
    alert_threshold: Optional[str] = None


class WatchResponseSchema(BaseModel):
    id: str
    name: str
    question: str
    aoi: dict
    sensor_preference: str
    frequency: str
    alert_threshold: Optional[str]
    status: str
    created_at: str
    updated_at: str
    last_run_at: Optional[str]
    next_run_at: Optional[str]

File: packages/types/src/watch.ts
export type SensorPreference = 'auto' | 'optical' | 'sar' | 'free';
export type WatchStatus = 'active' | 'paused' | 'error';
export type WatchFrequency = 'once' | 'daily' | 'weekly' | 'monthly';

export interface GeoJsonPolygon {
  type: 'Polygon';
  coordinates: number[][][];
}

export interface WatchCreate {
  name: string;
  question: string;
  aoi: GeoJsonPolygon;
  sensor_preference: SensorPreference;
  frequency: WatchFrequency;
  alert_threshold?: string;
}

export interface Watch extends WatchCreate {
  id: string;
  created_at: string;
  updated_at: string;
  status: WatchStatus;
  last_run_at?: string;
  next_run_at?: string;
}
